src/types.rs

```diff
@@ -1,13 +1,75 @@
+use core::{mem};
+use std::ops::{Deref, DerefMut};
+use std::{
+    convert::Infallible,
+    task::{Context, Poll},
+    pin::Pin,
+};
 use std::collections::HashMap;
 
 use actix_web::web::Bytes;
 use actix_web::{http::Method, HttpRequest};
+use actix_http::body::MessageBody;
+use actix_http::body::BodySize;
+
 use anyhow::Result;
 use dashmap::DashMap;
 use pyo3::{exceptions, prelude::*};
+use pyo3::types::PyBytes;
 
 use crate::io_helpers::read_file;
 
+
+#[derive(Debug, Clone)]
+#[pyclass]
+pub struct ActixBytesWrapper(Bytes);
+
+impl ActixBytesWrapper {
+    pub fn new(bytes: &PyBytes) -> Self {
+        Self(Bytes::from(bytes.as_bytes().to_vec()))
+    }
```

Comment:
Here is how you could handle both `bytes` and `str` from Python:
```suggestion
    pub fn new(raw_body: &PyAny) -> PyResult<Self> {
        let value = if let Ok(v) = raw_body.downcast::<PyString>() {
            v.to_string().into_bytes()
        } else if let Ok(v) = raw_body.downcast::<PyBytes>() {
            v.as_bytes().to_vec()
        } else {
            return Err(PyValueError::new_err(
                "Could not convert specified body to bytes",
            ));
        };
        Ok(Self(Bytes::from(value)))
    }
```

Reply:
❤️ Thank you, my knowledge of pyo3 is limited and knowing about this `downcast` is awesome.

Reply:
You're welcome, it's a pleasure to help 😉